jans-cache-refresh/src/main/java/io/jans/cacherefresh/service/AppInitializer.java

```diff
@@ -0,0 +1,300 @@
+
+		String disableScheduler = System.getProperties().getProperty("gluu.disable.scheduler");
+		if ((disableScheduler != null) && Boolean.valueOf(disableScheduler)) {
+			this.log.warn("Suspending Quartz Scheduler Service...");
+			quartzSchedulerManager.standby();
+			return;
+		}
+	}
+
+	@Produces
+	@ApplicationScoped
+	public StringEncrypter getStringEncrypter() {
+		String encodeSalt = configurationFactory.getCryptoConfigurationSalt();
+
+		if (StringHelper.isEmpty(encodeSalt)) {
+			throw new ConfigurationException("Encode salt isn't defined");
+		}
+
+		try {
+			StringEncrypter stringEncrypter = StringEncrypter.instance(encodeSalt);
+
+			return stringEncrypter;
+		} catch (EncryptionException ex) {
+			throw new ConfigurationException("Failed to create StringEncrypter instance");
+		}
+	}
+
+	protected Properties preparePersistanceProperties() {
+		PersistenceConfiguration persistenceConfiguration = this.configurationFactory.getPersistenceConfiguration();
+		FileConfiguration persistenceConfig = persistenceConfiguration.getConfiguration();
+		Properties connectionProperties = (Properties) persistenceConfig.getProperties();
+
+		EncryptionService securityService = encryptionServiceInstance.get();
+		Properties decryptedConnectionProperties = securityService.decryptAllProperties(connectionProperties);
+		return decryptedConnectionProperties;
+	}
+
+	protected Properties prepareCustomPersistanceProperties(String configId) {
+		Properties connectionProperties = preparePersistanceProperties();
+		if (StringHelper.isNotEmpty(configId)) {
+			// Replace properties names 'configId.xyz' to 'configId.xyz' in order to
+			// override default values
+			connectionProperties = (Properties) connectionProperties.clone();
+
+			String baseGroup = configId + ".";
+			for (Object key : connectionProperties.keySet()) {
+				String propertyName = (String) key;
+				if (propertyName.startsWith(baseGroup)) {
+					propertyName = propertyName.substring(baseGroup.length());
+
+					Object value = connectionProperties.get(key);
+					connectionProperties.put(propertyName, value);
+				}
+			}
+		}
+
+		return connectionProperties;
+	}
+
+	@Produces
+	@ApplicationScoped
+	@Named(ApplicationFactory.PERSISTENCE_ENTRY_MANAGER_NAME)
+	public PersistenceEntryManager createPersistenceEntryManager() {
+		Properties connectionProperties = preparePersistanceProperties();
+
+		PersistenceEntryManager persistenceEntryManager = applicationFactory.getPersistenceEntryManagerFactory()
+				.createEntryManager(connectionProperties);
+		log.info("Created {}: {} with operation service: {}",
+				new Object[] { ApplicationFactory.PERSISTENCE_ENTRY_MANAGER_NAME, persistenceEntryManager,
+						persistenceEntryManager.getOperationService() });
+
+		return persistenceEntryManager;
+	}
+
+	@Produces
+	@ApplicationScoped
+	@Named(ApplicationFactory.PERSISTENCE_METRIC_ENTRY_MANAGER_NAME)
+	@ReportMetric
+	public PersistenceEntryManager createMetricPersistenceEntryManager() {
```

Comment:
We can remove metric methods and classes because we not strart this service at all

Reply:
done